processor/schematransformerprocessor/config.go

```diff
@@ -0,0 +1,84 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schematransformerprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schematransformerprocessor"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+
+	"go.opentelemetry.io/collector/config"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schematransformerprocessor/internal/schema"
+)
+
+var (
+	errInvalidFileFormatVersion = errors.New("invalid file format version")
+	errInvalidSchemaURL         = errors.New("invalid URL schema")
+)
+
+// Config defines the user provided values for the Schema Transformer Processor
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"`
+
+	// SchemaURL defines where to download the published schema file
+	SchemaURL string `mapstructure:"schema_url"`
+
+	// SchemaFileVersion defines what the expected file format is
+	// and assist in making loading easier, this value isn't expected to change
+	// from the default is added know to make it aware that file versions matter.
+	SchemaFileVersion string `mapstructure:"schema_file_version"`
```

Comment:
This seems unnecessary. The file format version is already a field in the schema file. How does this setting here help with loading?

Reply:
If this field were to be used, it would remove the need to potential pass the bytes more than once since yaml formats don't enforce order of the dictionary.

For example, if versions was declared inside the file version then a rewind has to be done or reprocessing that map of type `interface`. 

This is where I was going with this however, it does seem like a micro optimisation since these files are only intended to be parsed once and then compiled.